conans/client/downloaders/cached_file_downloader.py

```diff
@@ -47,19 +48,27 @@ def download(self, url, file_path=None, md5=None, sha1=None, sha256=None, **kwar
                 if os.path.exists(cached_path):
                     os.remove(cached_path)
                 clean_dirty(cached_path)
-            if not os.path.exists(cached_path):
-                set_dirty(cached_path)
-                self._file_downloader.download(url=url, file_path=cached_path, md5=md5,
-                                               sha1=sha1, sha256=sha256, **kwargs)
-                clean_dirty(cached_path)
-            else:
-                # specific check for corrupted cached files, will raise, but do nothing more
-                # user can report it or "rm -rf cache_folder/path/to/file"
-                try:
-                    check_checksum(cached_path, md5, sha1, sha256)
-                except ConanException as e:
-                    raise ConanException("%s\nCached downloaded file corrupted: %s"
-                                         % (str(e), cached_path))
+            
+            for attempt in range(3):
+                if not os.path.exists(cached_path):
+                    set_dirty(cached_path)
+                    self._file_downloader.download(url=url, file_path=cached_path, md5=md5,
+                                                sha1=sha1, sha256=sha256, **kwargs)
+                    clean_dirty(cached_path)
```

Comment:
if successful, it shouldn't iterate, because ``file_downloader.download()`` already implements a checksum inside. Most likely a break is necessary here too.

Reply:
did not know that detail, done
```suggestion
                    clean_dirty(cached_path)
                    break
```